[onDeviceUI] Unifying React Native template.

Gongreg opened the pull request:
Part of #3903 

Currently there are two templates for React Native (vanilla and crna). Since we are 
unifying RN Storybook experience it makes sense to also unify Storybook installation.

Gongreg committed:
Unifying React Native template.

igor-dv reviewed (approved):
LGTM. I would mark it as breaking change though

Gongreg commented:
@igor-dv This one will impact new users only so I am not sure it counts as a breaking change.

Gongreg commented on lib/cli/lib/initiate.js:
```diff
@@ -96,26 +95,20 @@ export default function(options, pkg) {
-            logger.log(chalk.red('NOTE: CRNA app installation is not 100% automated.'));
-            logger.log(`To quickly run storybook, replace contents of ${app} with:\n`);
+            logger.log(chalk.red('NOTE: installation is not 100% automated.'));
+            logger.log(`To quickly run storybook, replace contents of your app entry with:\n`);
```
Add link to manual setup readme

Gongreg committed:
Updated instruction message in setup.
Added few comments in template.
Only notifies about not found app name if you use React Native Vanilla.

Gongreg commented:
Lets merge this together when merging #3903.

ndelangen committed:
Merge branch 'master' into react-native-cli-template
